tests/layer_tests/onnx_tests/test_loop.py

```diff
@@ -0,0 +1,401 @@
+            'Constant',
+            inputs=[],
+            outputs=['M_1'],
+            value=onnx.helper.make_tensor(
+                name='const_tensor',
+                data_type=onnx.TensorProto.INT64,
+                dims=m_1_value.shape,
+                vals=m_1_value.flatten().astype(np.int64),
+            ),
+        )
+
+        cond = onnx.helper.make_node(
+            'Constant',
+            inputs=[],
+            outputs=['cond'],
+            value=onnx.helper.make_tensor(
+                name='const_tensor',
+                data_type=onnx.TensorProto.BOOL,
+                dims=cond_value.shape,
+                vals=cond_value.flatten().astype(np.bool),
+            ),
+        )
+
+        one = onnx.helper.make_node(
+            'Constant',
+            inputs=[],
+            outputs=['one'],
+            value=onnx.helper.make_tensor(
+                name='const_tensor',
+                data_type=onnx.TensorProto.FLOAT,
+                dims=one_value.shape,
+                vals=one_value.flatten().astype(np.float),
+            ),
+        )
+
+        one_int = onnx.helper.make_node(
+            'Constant',
+            inputs=[],
+            outputs=['one_int'],
+            value=onnx.helper.make_tensor(
+                name='const_tensor',
+                data_type=onnx.TensorProto.INT64,
+                dims=one_value.shape,
+                vals=one_value.flatten().astype(np.int64),
+            ),
+        )
+
+        add_one_node = onnx.helper.make_node(
+            'Add',
+            inputs=['in_1_int', 'one'],
+            outputs=['OUT_1']
+        )
+
+        add_one_to_m_node = onnx.helper.make_node(
+            'Add',
+            inputs=['m_1', 'one_int'],
+            outputs=['m_1_loop_1']
+        )
+
+        cond_1_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['cond_int_1'],
+            outputs=['cond_out_1'],
+        )
+
+        in_1_int_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['in_1_int'],
+            outputs=['in_1_int_out'],
+        )
+
+        body_graph_1 = helper.make_graph(
+            [one, one_int, add_one_node, add_one_to_m_node, cond_1_node, in_1_int_node],
+            'body_graph_1',
+            [m_1, cond_int_1, in_1_int],
+            [cond_out_1, in_1_int_out, out_1],
+        )
+
+        node_loop_1 = onnx.helper.make_node(
+            'Loop',
+            inputs=['M_1', 'cond', 'IN_1'],
+            outputs=['cond_out_1', 'OUT_1'],
+            body=body_graph_1
+        )
+
+        res_node = onnx.helper.make_node(
+            'Identity',
+            inputs=['OUT_1'],
+            outputs=['res'],
+        )
+
+        graph_def = helper.make_graph(
+            [M_1, cond, node_loop_1, res_node],
+            'graph',
+            [in_1],
+            [res],
+        )
+
+        onnx_net = helper.make_model(graph_def, producer_name='test_loop_model')
```

Comment:
I propose to think of implementing a function that creates body graph 1 and will reuse it in both tests (Loop(body graph #1)) and Loop(body graph #2 with Loop(body graph #1)).

Reply:
also please add comments for each step where you create body graph and the main graph